components/builder-api/src/server/resources/channels.rs

```diff
@@ -444,10 +444,12 @@ async fn promote_package(req: HttpRequest,
                 &auditevent,
                 &*conn,
             ) {
-                Ok(_) => publish_event!(&state, EventType::PackageChannelMotion, json!(&auditevent)),
+                Ok(_) => {}
```

Comment:
This empty arm suggests that you could get away with something like this instead:

```rust
if let Err(e) = PackageChannelAudit::audit(&auditevent, &*conn) {
    debug!("Failed to save rank change to audit log: {}", e);
}
```

Reply:
Agreed. Handled in https://github.com/habitat-sh/builder/pull/1540/commits/9f5f1e41cf492cd955da60b6f382128239b3a073